spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/annotation/SqsListenerAnnotationBeanPostProcessor.java

```diff
@@ -0,0 +1,260 @@
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * {@link BeanPostProcessor} implementation that scans beans for a {@link SqsListener @SqsListener} annotation, extracts
+ * information to a {@link SqsEndpoint}, and registers it in the {@link EndpointRegistrar}.
+ *
+ * @author Tomaz Fernandes
+ * @since 3.0
+ */
+public class SqsListenerAnnotationBeanPostProcessor
+		implements BeanPostProcessor, BeanFactoryAware, SmartInitializingSingleton {
+
+	private static final String GENERATED_ID_PREFIX = "io.awspring.cloud.sqs.sqsListenerEndpointContainer#";
+
+	private final AtomicInteger counter = new AtomicInteger();
+
+	private final Collection<Class<?>> nonAnnotatedClasses = Collections.synchronizedSet(new HashSet<>());
+
+	private final ExpressionResolvingHelper expressionResolvingHelper = createExpressionResolvingHelper();
+
+	private final EndpointRegistrar endpointRegistrar = createEndpointRegistrar();
+
+	private final DelegatingMessageHandlerMethodFactory delegatingHandlerMethodFactory = new DelegatingMessageHandlerMethodFactory();
+
+	private BeanFactory beanFactory;
+
+	@Override
+	public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+		Class<?> targetClass = AopUtils.getTargetClass(bean);
+		if (this.nonAnnotatedClasses.contains(targetClass)) {
+			return bean;
+		}
+		detectAnnotationsAndRegisterEndpoints(bean, targetClass);
+		return bean;
+	}
+
+	protected ExpressionResolvingHelper resolveExpression() {
+		return this.expressionResolvingHelper;
+	}
+
+	@Nullable
+	protected ConfigurableBeanFactory getConfigurableBeanFactory() {
+		return this.beanFactory instanceof ConfigurableBeanFactory ? (ConfigurableBeanFactory) this.beanFactory : null;
+	}
+
+	protected BeanFactory getBeanFactory() {
+		return this.beanFactory;
+	}
+
+	protected void detectAnnotationsAndRegisterEndpoints(Object bean, Class<?> targetClass) {
+		Map<Method, SqsListener> annotatedMethods = MethodIntrospector.selectMethods(targetClass,
+				(MethodIntrospector.MetadataLookup<SqsListener>) method -> AnnotatedElementUtils
+						.findMergedAnnotation(method, SqsListener.class));
+		if (annotatedMethods.isEmpty()) {
+			this.nonAnnotatedClasses.add(targetClass);
+		}
+		annotatedMethods.entrySet().stream()
+				.map(entry -> createEndpointFromAnnotation(bean, entry.getKey(), entry.getValue()))
+				.forEach(this.endpointRegistrar::registerEndpoint);
+	}
+
+	private Endpoint createEndpointFromAnnotation(Object bean, Method method, SqsListener annotation) {
+		Endpoint endpoint = doCreateEndpointFromAnnotation(annotation);
+		ConfigUtils.INSTANCE.acceptIfInstance(endpoint, HandlerMethodEndpoint.class, hme -> {
+			hme.setBean(bean);
+			hme.setMethod(method);
+			hme.setHandlerMethodFactory(this.delegatingHandlerMethodFactory);
+		});
+		return endpoint;
+	}
+
+	private Endpoint doCreateEndpointFromAnnotation(SqsListener sqsListenerAnnotation) {
```

Comment:
just `createEndpoint`?

Reply:
I've changed the names to `createAndConfigureEndpoint` and `createEndpoint` - let me know if that doesn't work for you.